### PR TITLE
Migrate from GitLab CI to GitHub Actions

### PR DESCRIPTION
**Pull request by pgrete**
## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [ ] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Comment by pgrete** [comment]
This is ready for review.
As far as I can tell, the nightly and triggered extended workflow can only be tested once this PR is merged in as the default target branch of the extended flow is `main`, which currently doesn't know about the workflow.

This PR now also enables the action to run on forked repos (upon manual approve).

**Review by Yurlungur — APPROVED** [review]
:+1: big fan of this change. From my perspective one review is enough for this change, so go ahead and merge at your convenience.

**Comment by Yurlungur** [comment]
(I'd merge myself but I can't because I'm not a maintainer and can't force merge, given the failed gitlab tests)

File: .github/workflows/ci-short.yml
Comment (lroberts36) [inline .github/workflows/ci-short.yml:105]: The success or failure of this test seems to be sensitive to the hardware it is running on, since `--delay=5` and `--duration=5` may pick up the final step, which seems to involve HtoD and DtoH copies.

**Review by lroberts36 — APPROVED** [review]
This LGTM. Thanks for making the switch!

File: .github/workflows/ci-short.yml
Comment (pgrete) [reply .github/workflows/ci-short.yml:105]: Correct. This is specifically tuned to the CI machine and would need to be adjusted once we migrate.
I'll add a comment next month (when I do the migration to a machine in Hamburg).

**Comment by pgrete** [comment]
@jdolence can you force-merge? I don't know how to remove the GitLab status check and don't know why the formatting check doesn't run.

**Comment by pgrete** [comment]
@Yurlungur @lroberts36, is @jdolence on vacation?
Might be worth to expand the group of people who can manage the repo.

**Comment by Yurlungur** [comment]
> @Yurlungur @lroberts36, is @jdolence on vacation? Might be worth to expand the group of people who can manage the repo.

No he's here, probably missed the first ping. I just pinged him.

**Pull request merged**
